pkg/registry/storage/csinode/strategy.go

```diff
@@ -54,16 +62,21 @@ func (csiNodeStrategy) Validate(ctx context.Context, obj runtime.Object) field.E
 	return errs
 }
 
-// Canonicalize normalizes the object after validation.
 func (csiNodeStrategy) Canonicalize(obj runtime.Object) {
 }
 
 func (csiNodeStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
-// PrepareForUpdate sets the Status fields which is not allowed to be set by an end user updating a CSINode
+// PrepareForUpdate clears fields that are not allowed to be set on update.
 func (csiNodeStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	csiNode := obj.(*storage.CSINode)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.AttachVolumeLimit) {
+		for _, csiDriver := range csiNode.Spec.Drivers {
+			csiDriver.Allocatable = nil
```

Comment:
The guidance we've been following for all other feature-gated fields, is that if it's already set, we continue to let it be set.  This handles cases where the feature was on, and then the cluster was downgraded to a version where the feature is off.

Reply:
SGTM. Removed